### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by AckonSamuel**
# In this PR, these were the changes implemented:

- [x] Remove the current_user method in ApplicationController
- [x] Install and setup devise
- [x] Allow the app to :
- [x] Accept login and registration
- [x] Login with mail and password
- [x] Sign In as a new user
- [x] Perform all functionalities for a user while logged in
- [x] Modified the registrations and the login views to match the wireframes

## NOTE TO REVIEWER
Please, kindly open the HTML file to view the confirmation link. Your browser may open it automatically if it permits you to open HTML files. You can also find the HTML file in the ` /tmp/letter_opener/` directory.
please check the address in the image
image

### CODE REVIEW

File: db/migrate/20221104112658_add_more_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddMoreToUsers < ActiveRecord::Migration[7.0]
```
Comment (SajjadAhmad14) [inline db/migrate/20221104112658_add_more_to_users.rb:1]: There is a migration error so I could not run the app. Kindly make sure the database is created and migrated successfully.
mig

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Samuel, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20221104112658_add_more_to_users.rb (hunk 1 above)
Comment (AckonSamuel) [reply db/migrate/20221104112658_add_more_to_users.rb:1]: okay thank you so much 😄

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @AckonSamuel, 

Great work on making the changes requested by a previous reviewer 👏 🔥
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights 🎉🔥
- Removed current_user method ✔
- Use devise actions ✔
- No linter error ✔

 ## Required Changes ♻️
- [x] There is an issue after signing up, however, the confirmation mail is sent as expected! Let's fix the registration action
- [Update for the next reviewer] The issue shown in the picture below may be optional.

Screen Shot 2022-11-04 at 4 00 11 PM

_Check the comments under the review._

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AckonSamuel** [comment]
@EricMbouwe thank you for the review. I have fixed the button issue 😄

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @AckonSamuel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**